framework/src/main/java/org/checkerframework/framework/ajava/InsertAjavaAnnotations.java

```diff
@@ -0,0 +1,589 @@
+    insertions.sort(InsertAjavaAnnotations::compareInsertions);
+
+    StringBuilder result = new StringBuilder(javaFileContents);
+    for (Insertion insertion : insertions) {
+      result.insert(insertion.position, insertion.contents);
+    }
+    return result.toString();
+  }
+
+  /**
+   * Compares two insertions in the reverse order of where their content should appear in the file.
+   * Making an insertion changes the offset values of all content after the insertion, so performing
+   * the insertions in reverse order of appearance removes the need to recalculate the positions of
+   * other insertions.
+   *
+   * <p>The order in which insertions should appear is determined first by their absolute position
+   * in the file, and second by whether they have their own line. In a method like
+   * {@code @Pure @Tainting String myMethod()} both annotations should be inserted at the same
+   * location (right before "String"), but {@code @Pure} should always come first because it belongs
+   * on its own line.
+   *
+   * @param insertion1 the first insertion
+   * @param insertion2 the second insertion
+   * @return a negative integer, zero, or a positive integer if {@code insertion1} belongs before,
+   *     at the same position, or after {@code insertion2} respectively in the above ordering
+   */
+  private static int compareInsertions(Insertion insertion1, Insertion insertion2) {
+    int cmp = Integer.compare(insertion1.position, insertion2.position);
+    if (cmp == 0 && (insertion1.ownLine != insertion2.ownLine)) {
+      if (insertion1.ownLine) {
+        cmp = -1;
+      } else {
+        cmp = 1;
+      }
+    }
+
+    return -cmp;
+  }
+
+  /**
+   * Inserts all annotations from the ajava file at {@code annotationFilePath} into {@code
+   * javaFilePath}.
+   *
+   * @param annotationFilePath path to an ajava file
+   * @param javaFilePath path to a Java file to insert annotation into
+   */
+  public void insertAnnotations(String annotationFilePath, String javaFilePath) {
+    try {
+      Path path = Paths.get(javaFilePath);
+      String fileContents = new String(Files.readAllBytes(path));
+      FileInputStream annotationInputStream = new FileInputStream(annotationFilePath);
+      String result = insertAnnotations(annotationInputStream, fileContents);
+      annotationInputStream.close();
+      Files.write(path, result.getBytes());
+    } catch (IOException e) {
+      System.err.println(
+          "Failed to insert annotations from file "
+              + annotationFilePath
+              + " into file "
+              + javaFilePath);
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Inserts annotations from ajava files into Java files in place.
+   *
+   * <p>The first argument is an ajava file or a directory containing ajava files.
+   *
+   * <p>The second argument is a Java file or a directory containing Java files to insert
+   * annotations into. The files must use the same line separator as the host system.
```

Comment:
Previously, it was important for correctness because how the lines were broken up affected whether we decided to put an annotation on its own line. But it's moot now since I switched it to infer the line separator, as you suggested.